File: plugins/capawesome-capacitor-wifi/android/src/main/java/io/capawesome/capacitorjs/plugins/wifi/classes/results/GetSsidResult.java
package io.capawesome.capacitorjs.plugins.wifi.classes.results;

import android.net.wifi.WifiInfo;
import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.wifi.interfaces.Result;

public class GetSsidResult implements Result {

    private String ssid;

    public GetSsidResult(@NonNull WifiInfo wifiInfo) {
        this.ssid = wifiInfo.getSSID().substring(1, wifiInfo.getSSID().length() - 1);
    }

    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("ssid", ssid);
        return result;
    }
}

File: plugins/capawesome-capacitor-wifi/ios/Plugin/Classes/Results/GetSsidResult.swift
import Foundation
import CoreBluetooth
import Capacitor

@objc public class GetSsidResult: NSObject, Result {
    private let ssid: String

    init(ssid: String) {
        self.ssid = ssid
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["ssid"] = ssid
        return result as AnyObject
    }
}
